lib/rmt/scc.rb

```diff
@@ -185,9 +203,24 @@ def create_product(item, root_product_id = nil, base_product = nil, recommended
   end
 
   def create_service(item, product)
-    product.create_service!
+    product.find_or_create_service!
+    existing_repo_ids = product.repositories.only_scc.pluck(:scc_id)
+
     item[:repositories].each do |repo_item|
-      repository_service.create_repository!(product, repo_item[:url], repo_item)
+      repository_service.update_or_create_repository!(product, repo_item[:url], repo_item)
+      existing_repo_ids.delete(repo_item[:id])
+    end
+    disassociate_obsolete_repositories(product, existing_repo_ids)
+  end
+
+  def disassociate_obsolete_repositories(product, existing_repo_ids)
```

Comment:
This reads pretty complicated, would it be possible to do something like: 

```
repository_service.disassociate_repositories(product, 
  (service.repositories.only_scc - product.repositories.only_scc))
```
wdyt?